fe/fe-core/src/main/java/com/starrocks/common/util/ExpressionPartitionUtil.java

```diff
@@ -0,0 +1,219 @@
+// This file is licensed under the Elastic License 2.0. Copyright 2021-present, StarRocks Limited.
+package com.starrocks.common.util;
+
+import com.google.common.collect.Maps;
+import com.google.common.collect.Range;
+import com.starrocks.analysis.Expr;
+import com.starrocks.analysis.FunctionCallExpr;
+import com.starrocks.analysis.PartitionValue;
+import com.starrocks.analysis.SlotRef;
+import com.starrocks.analysis.StringLiteral;
+import com.starrocks.catalog.Column;
+import com.starrocks.catalog.PartitionKey;
+import com.starrocks.catalog.PrimitiveType;
+import com.starrocks.common.AnalysisException;
+import com.starrocks.common.DdlException;
+import com.starrocks.sql.analyzer.SemanticException;
+import com.starrocks.sql.optimizer.operator.scalar.ConstantOperator;
+import com.starrocks.sql.optimizer.rewrite.ScalarOperatorFunctions;
+
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Map;
+
+/**
+ * Process lower border and upper border for Expression Partition,
+ * only support SlotRef and FunctionCallExpr
+ */
+public class ExpressionPartitionUtil {
```

Comment:
Would better move this class to com.starrocks.sql.common. The basic common class shouldn't know the expression.